extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -85,27 +85,44 @@ public void initialize() {
             return;
         }
 
-        this.gatewayURI = "coaps://" + configuration.host + ":" + configuration.port + "/" + DEVICES;
+        DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
+        builder.setPskStore(new StaticPskStore("", configuration.code.getBytes()));
+        dtlsConnector = new DTLSConnector(builder.build());
+        endPoint = new CoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
+
+        // setup DEVICES scanner
+        devices = new MyCoapCallback();
+        devices.gatewayURI = "coaps://" + configuration.host + ":" + configuration.port + "/" + DEVICES;
         try {
-            URI uri = new URI(gatewayURI);
-            deviceClient = new TradfriCoapClient(uri);
+            URI uri = new URI(devices.gatewayURI);
+            devices.client = new TradfriCoapClient(uri);
         } catch (URISyntaxException e) {
-            logger.debug("Illegal gateway URI `{}`: {}", gatewayURI, e.getMessage());
+            logger.debug("Illegal gateway URI `{}`: {}", devices.gatewayURI, e.getMessage());
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
             return;
         }
+        devices.client.setEndpoint(endPoint);
+
+        // setup GROUPS scanner
```

Comment:
As this more or less duplicates the code from above, could you please factor out the common parts into a separate method?